provider/pkg/gen/schema.go

```diff
@@ -518,7 +518,7 @@ Use the navigation below to see detailed documentation for each of the supported
 	})
 	pkg.Language["python"] = rawMessage(map[string]any{
 		"requires": map[string]string{
-			"pulumi":   ">=3.25.0,<4.0.0",
+			"pulumi":   ">=3.109.0,<4.0.0",
```

Comment:
Alternatively, instead of bumping the min required version, we could dynamically check whether `invoke_async` exists in `pulumi.runtime`, and fallback to the old implementations if it's unavailable. That does add some complexity, though.